internal/manifests/gateway.go

```diff
@@ -0,0 +1,184 @@
+package manifests
+
+import (
+	"fmt"
+	"path"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ViaQ/loki-operator/internal/manifests/internal/gateway"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+// BuildLokiStackGateway returns a list of k8s objects for Loki Stack Gateway
+func BuildLokiStackGateway(opts Options) ([]client.Object, error) {
+	gatewayConfigMap, err := GatewayConfigMap(opts)
+	if err != nil {
+		return nil, err
+	}
+	return []client.Object{
+		gatewayConfigMap,
+		NewLokiStackGatewayDeployment(opts),
+	}, nil
+}
+
+// NewLokiStackGatewayDeployment creates a deployment object for a lokiStack-gateway
+func NewLokiStackGatewayDeployment(opts Options) *appsv1.Deployment {
+	podSpec := corev1.PodSpec{
+		Volumes: []corev1.Volume{
+			{
+				Name: "rbac",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: labelObservatoriumApi,
+						},
+					},
+				},
+			},
+			{
+				Name: "tenants",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: labelObservatoriumApi,
+						},
+					},
+				},
+			},
+		},
+		Containers: []corev1.Container{
+			{
+				Name:  LabelLokiStackGatewayComponent,
+				Image: DefaultLokiStackGatewayImage,
+				Args: []string{
+					fmt.Sprintf("--debug.name=%s", LabelLokiStackGatewayComponent),
+					"--web.listen=0.0.0.0:8080",
+					"--web.internal.listen=0.0.0.0:8081",
+					"--log.level=debug",
+					fmt.Sprintf("--logs.read.endpoint=http://%s:%d", fqdn(serviceNameQueryFrontendHTTP(opts.Name), opts.Namespace), httpPort),
+					fmt.Sprintf("--logs.tail.endpoint=http://%s:%d", fqdn(serviceNameQuerierHTTP(opts.Name), opts.Namespace), httpPort),
```

Comment:
We can use the query frontend for tails too nowdays. It was pre-loki-2.2 a limitation.